File: api/src/main/java/com/clinics_schedules/clinic_api/dto/ClinicScheduleDto.java
package com.clinics_schedules.clinic_api.dto;

import java.time.LocalTime;
import java.util.List;

import com.clinics_schedules.clinic_api.entity.ClinicSchedule;
import com.clinics_schedules.clinic_api.entity.Employee;
import com.clinics_schedules.clinic_api.enums.TimeRepeatUnit;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.Setter;
import lombok.experimental.Accessors;

@Getter
@Setter
@AllArgsConstructor
@Accessors(chain = true)
public class ClinicScheduleDto {
    private Integer id;
    private Integer clinicId;
    private Long beginDate;
    private Long expireDate;
    private LocalTime eventStart;
    private LocalTime eventFinish;
    private TimeRepeatUnit repeat;

    private List<EventDto> events;
    private List<Employee> employees;

    public ClinicScheduleDto(final ClinicSchedule schedule) {
        this.id = schedule.getId();
        this.clinicId = schedule.getClinicId();
        this.beginDate = schedule.getBeginDate().getTime();
        this.expireDate = schedule.getExpireDate().getTime();
        this.eventStart = schedule.getEventStart();
        this.eventFinish = schedule.getEventFinish();
        this.repeat = schedule.getRepeat();
        this.employees = schedule.getEmployees();
    }

}

File: frontend/src/app/dto/ClinicScheduleDto.ts
import { EventDto } from "./EventDto";
import { EmployeeDto } from "./EmployeeDto";

export type RepeatUnit = 'never' | 'daily' | 'weekly' | "weekdays" | "weekend";
export class ClinicScheduleDto {

    constructor(
        public id: number,
        public clinicId: number,
        public clinicName: string | undefined,
        public beginDate: string | number,
        public expireDate: string | number,
        public eventStart: string,
        public eventFinish: string,
        public repeat: RepeatUnit,
        public events: EventDto[],
        public employees: EmployeeDto[],
        public note: string,
    ) {

    }
}
